core/src/main/java/org/infinispan/atomic/impl/AtomicHashMap.java

```diff
@@ -254,4 +277,9 @@ public Integer getId() {
          return Util.<Class<? extends AtomicHashMap>>asSet(AtomicHashMap.class);
       }
    }
+
+   public enum ProxyMode {
+      NONE, FINE, COARSE;
+      public static final ProxyMode[] CACHED_VALUES = values();
```

Comment:
The same comment as for https://github.com/infinispan/infinispan/pull/3930, having this field public is confusing.